sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -329,7 +332,17 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * @since 1.4.0
    */
   def json(jsonRDD: RDD[String]): DataFrame = {
-    val parsedOptions: JSONOptions = new JSONOptions(extraOptions.toMap)
+    val optionsWithTimeZone = {
```

Comment:
The `timeZone` option is used in the `JSONOptions`/`CSVOptions`, so we can't handle it the same as `columnNameOfCorruptRecord`.
I'll modify to pass the default timezone id to `JSONOptions` and `CSVOptions`.